src/script/sign.cpp

```diff
@@ -149,8 +149,9 @@ bool ProduceSignature(const BaseSignatureCreator &creator, const CScript &fromPu
     }
 
     // Test solution
-    return VerifyScript(
-        scriptSig, fromPubKey, STANDARD_SCRIPT_VERIFY_FLAGS | SCRIPT_ENABLE_SIGHASH_FORKID, creator.Checker());
+    // We can hard-code maxOps because this client has not templates capable of producing and signing longer scripts.
+    return VerifyScript(scriptSig, fromPubKey, STANDARD_SCRIPT_VERIFY_FLAGS | SCRIPT_ENABLE_SIGHASH_FORKID,
+        MAX_OPS_PER_SCRIPT, creator.Checker());
```

Comment:
I don't want to pull maxScriptOps into this lower level code, although it is possible by making it a unsigned int and using TweakRef.  

This wallet cannot sign any tx that get anywhere near the max ops anyway because it is only capable of signing the standard transactions.  I don't see this changing anytime soon, so I chose to keep-it-simple by using the constant (esp since having it configurable is likely transitional -- it'll be a hard coded constant, maybe 0xffffffff -- effectively infinite -- once the scalability work has been done).

Reply:
wonder if it would be worth adding a comment explaining the choice that have been made.

Reply:
+1